datatypes/src/dataset.rs

```diff
@@ -0,0 +1,62 @@
+use crate::identifier;
+use serde::{Deserialize, Serialize};
+
+identifier!(DataSetProviderId);
+
+identifier!(InternalDataSetId);
+
+identifier!(StagingDataSetId);
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq, Deserialize, Serialize)]
```

Comment:
> ChristianBeilschmidt commented on 18.01.2021
How is this serialized? I guess we don't want to say:
{
	"operator": "source",
	"dataset": {
		"Internal": "abcd-1234-abcd-1234"
	}
}

> jdroenner commented on 19.01.2021
the alternative is "internal:abcd-1234-abcd-1234". I like the prefix / KV as it enables us to have different handlers.

I don't see a problem with the native serialization but we could always change it if there is an advantage

Reply:
Can't we just have UUIDs and the system resolves if it is internal or is it advantageous to define it in the workflow?

Reply:
possible advantages:
- wokflow reflects where the data resides without looking up further meta data
- implementation allows requiring a specific type of id, e.g. `fn import` requires a `StagedId`
- saves the extra step to lookup the location of the data set because it is encoded in the id

on the other hand one id type makes for a cleaner interface.

Reply:
i still think it would be a good idea to be able to add more prefixes. I would suggest to leave it as it is.